sources/oracle/infoschema.go

```diff
@@ -29,21 +43,92 @@ func (isi InfoSchemaImpl) GetTableName(dbName string, tableName string) string {
 
 // GetRowsFromTable returns a sql Rows object for a table.
 func (isi InfoSchemaImpl) GetRowsFromTable(conv *internal.Conv, srcTable string) (interface{}, error) {
-	panic("Not Implemented")
+	tbl := conv.SrcSchema[srcTable]
+	srcCols := tbl.ColNames
+	if len(srcCols) == 0 {
+		conv.Unexpected(fmt.Sprintf("Couldn't get source columns for table %s ", srcTable))
+		return nil, nil
+	}
+	q := getSelectQuery(isi.DbName, tbl.Schema, tbl.Name, tbl.ColNames, tbl.ColDefs)
+	rows, err := isi.Db.Query(q)
+	return rows, err
 }
 
-// ProcessData performs data conversion for source database.
-func (isi InfoSchemaImpl) ProcessData(conv *internal.Conv, srcTable string, srcSchema schema.Table, spTable string, spCols []string, spSchema ddl.CreateTable) error {
-	panic("Not Implemented")
+func getSelectQuery(srcDb string, schemaName string, tableName string, colNames []string, colDefs map[string]schema.Column) string {
+	var selects = make([]string, len(colNames))
+
+	for i, cn := range colNames {
+		var s string
+		if TimestampReg.MatchString(colDefs[cn].Type.Name) {
+			s = fmt.Sprintf(`SYS_EXTRACT_UTC("%s") AS "%s"`, cn, cn)
+		} else if len(colDefs[cn].Type.ArrayBounds) == 1 {
+			s = fmt.Sprintf(`(SELECT JSON_ARRAYAGG(COLUMN_VALUE RETURNING VARCHAR2(4000)) 
+				FROM TABLE ("%s"."%s")) AS "%s"`, tableName, cn, cn)
+		} else {
+			switch colDefs[cn].Type.Name {
+			case "XMLTYPE":
+				s = fmt.Sprintf(`CAST(XMLTYPE.getStringVal("%s") AS VARCHAR2(4000)) AS "%s"`, cn, cn)
+			case "SDO_GEOMETRY":
+				s = fmt.Sprintf(`SDO_UTIL.TO_WKTGEOMETRY("%s") AS "%s"`, cn, cn)
+			case "OBJECT":
+				s = fmt.Sprintf(`
+				(
+					CASE WHEN "%s" IS NULL THEN ''
+					ELSE
+						XMLTYPE("%s").getStringVal() 
+					END
+				) AS "%s"
+				`, cn, cn, cn)
+			default:
+				s = fmt.Sprintf(`"%s"`, cn)
+			}
+		}
+		selects[i] = s
+	}
+
+	return fmt.Sprintf(`SELECT %s FROM "%s"."%s"`, strings.Join(selects, ", "), schemaName, tableName)
```

Comment:
Column names can have special characters and can be lower/upper case. They need to be quoted - https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements008.htm

Reply:
%s is here is a joined string of quoted table names. while populating "selects" we added table names within double quotes.